internal/cri/server/images/image_pull.go

```diff
@@ -133,6 +133,20 @@ func (c *CRIImageService) PullImage(ctx context.Context, name string, credential
 	defer inProgressImagePulls.Dec()
 	startTime := time.Now()
 
+	if credentials == nil {
```

Comment:
IMO, it might be better to generate `credentials` in this function. It's easy to track and follow.

https://github.com/containerd/containerd/blob/6b6f53cb1e0bac29d9fc8792266ed56a684a012e/internal/cri/server/podsandbox/sandbox_run.go#L302

Reply:
if you could explain to me how, I'm happy to do that, it seems that podsanbdox service doesn't have access to the config of the image service, so it can't pull the creds. 

(I'm not an expert in containerd architecture, so please correct me if I'm wrong)

Reply:
You're right :)